crates/redis_interface/src/commands.rs

```diff
@@ -280,20 +285,24 @@ impl super::RedisConnectionPool {
         V: TryInto<RedisValue> + Debug + Send + Sync,
         V::Error: Into<fred::error::RedisError> + Send + Sync,
     {
-        let output: Result<HsetnxReply, _> = self
-            .pool
+        let pipeline = self.pool.next().pipeline();
+        let _hset_result: RedisValue = pipeline
             .hsetnx(key, field, value)
             .await
             .into_report()
-            .change_context(errors::RedisError::SetHashFieldFailed);
-
-        output
-            .async_and_then(|inner| async {
-                self.set_expiry(key, ttl.unwrap_or(self.config.default_hash_ttl).into())
-                    .await?;
-                Ok(inner)
-            })
+            .change_context(errors::RedisError::SetHashFieldFailed)?;
+        let _expiry_result: String = pipeline
+            .expire(key, ttl.unwrap_or(self.config.default_hash_ttl).into())
             .await
+            .into_report()
+            .change_context(errors::RedisError::SetExpiryFailed)?;
+        let results = pipeline.try_all::<RedisValue>().await;
+
+        results[0]
```

Comment:
result[0] is a unsafe to get a value out of vec's. Use functions which return Option so that we can handle the null case.